src/test/java/bio/terra/integration/IngestTest.java

```diff
@@ -343,4 +343,220 @@ public void ingestWildcardMalformedTest() throws Exception {
+    IngestResponseModel ingestResponse =
+        dataRepoFixtures.ingestJsonData(steward(), datasetId, ingestRequest);
+    assertThat("correct sample row count", ingestResponse.getRowCount(), equalTo(7L));
+
+    IngestRequestModel mergeIngestRequest =
+        dataRepoFixtures
+            .buildSimpleIngest(
+                "sample", "ingest-test/merge/ingest-test-sample-merge-missing-pks.json")
+            .updateStrategy(UpdateStrategyEnum.MERGE);
+    DataRepoResponse<JobModel> mergeIngestJobResponse =
+        dataRepoFixtures.ingestJsonDataLaunch(steward(), datasetId, mergeIngestRequest);
+    DataRepoResponse<IngestResponseModel> mergeIngestResponse =
+        dataRepoClient.waitForResponse(steward(), mergeIngestJobResponse, new TypeReference<>() {});
+
+    assertThat(
+        "ingest failed", mergeIngestResponse.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+    assertThat(
+        "failure is explained",
+        mergeIngestResponse.getErrorObject().get().getMessage(),
+        containsString("Ingest failed"));
+    assertThat(
+        "primary key specification is enforced",
+        mergeIngestResponse.getErrorObject().get().getErrorDetail(),
+        hasItem(containsString("Missing required field: id")));
+  }
+
+  @Test
+  public void ingestMergeRowsDuplicatePKsTest() throws Exception {
+    IngestRequestModel ingestRequest =
+        dataRepoFixtures.buildSimpleIngest("sample", "ingest-test/ingest-test-sample.json");
+    IngestResponseModel ingestResponse =
+        dataRepoFixtures.ingestJsonData(steward(), datasetId, ingestRequest);
+    assertThat("correct sample row count", ingestResponse.getRowCount(), equalTo(7L));
+
+    IngestRequestModel mergeIngestRequest =
+        dataRepoFixtures
+            .buildSimpleIngest(
+                "sample", "ingest-test/merge/ingest-test-sample-merge-duplicate-pks.json")
+            .updateStrategy(UpdateStrategyEnum.MERGE);
+    DataRepoResponse<JobModel> mergeIngestJobResponse =
+        dataRepoFixtures.ingestJsonDataLaunch(steward(), datasetId, mergeIngestRequest);
+    DataRepoResponse<IngestResponseModel> mergeIngestResponse =
+        dataRepoClient.waitForResponse(steward(), mergeIngestJobResponse, new TypeReference<>() {});
+
+    assertThat(
+        "ingest failed", mergeIngestResponse.getStatusCode(), equalTo(HttpStatus.BAD_REQUEST));
+    assertThat(
+        "failure is explained",
+        mergeIngestResponse.getErrorObject().get().getMessage(),
+        containsString("Duplicate primary keys identified"));
+    assertThat(
+        "all duplicate primary keys are found in error details",
+        mergeIngestResponse.getErrorObject().get().getErrorDetail(),
+        containsInAnyOrder(containsString("sample1")));
+  }
+
+  @Test
+  public void ingestMergeMismatchedWithTargetTest() throws Exception {
+    IngestRequestModel ingestRequest =
+        dataRepoFixtures.buildSimpleIngest(
+            "participant", "ingest-test/ingest-test-participant.json");
```

Comment:
Did you mean to create an `ingest-test` subdirectory as well? These paths don't match up with the test files

Reply:
These paths match the paths in cloud storage, which don't exactly match the paths in source control :( I was reluctant to move around existing files in cloud storage and/or source control in case I accidentally broke tests.

Here's a direct link:

https://console.cloud.google.com/storage/browser/jade-testdata/ingest-test

Reply:
That sounds like a good compromise!  I can also make a note of this in the ticket I'll write up for hardening the connection between GCS / Git test resources.  (And maybe work coming from that ticket will make this all a moot point!)

Reply:
@samanehsan please let me know if my latest change is an acceptable compromise.